src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/ExecuteActionTests.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options.Actions;
+using System.Threading.Tasks;
+using Xunit;
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Actions;
+using System.Reflection;
+using Microsoft.Diagnostics.Monitoring.TestCommon;
+using System.Threading;
+using System;
+using System.IO;
+using System.Diagnostics;
+using Microsoft.Diagnostics.Tools.Monitor;
+
+namespace Microsoft.Diagnostics.Monitoring.Tool.UnitTests
+{
+    public sealed class ExecuteActionTests
+    {
+        private const int TokenTimeoutMs = 2000; // Must be less than DelayMs in the ExecuteApp
```

Comment:
This timeout needs to be much longer, more like 10 seconds, to account for PR/CI build slowness.